sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java

```diff
@@ -84,4 +86,57 @@ void fromScheduledTransaction() {
 
         assertThat(tx).isInstanceOf(TokenAssociateTransaction.class);
     }
+
+    @Test
+    void ConstructTokenDeleteTransactionFromTransactionBodyProtobuf() {
+        var accountId = AccountId.fromString("1.2.3");
+        var tokenIds = List.of(TokenId.fromString("4.5.6"),
+            TokenId.fromString("7.8.9"),
+            TokenId.fromString("10.11.12"));
+
+        var transactionBody = TokenAssociateTransactionBody.newBuilder()
+            .addAllTokens(tokenIds.stream().map(TokenId::toProtobuf).toList())
+            .setAccount(accountId.toProtobuf()).build();
+        var txBody = TransactionBody.newBuilder().setTokenAssociate(transactionBody).build();
+        var tokenAssociateTransaction = new TokenAssociateTransaction(txBody);
+
+        assertThat(tokenAssociateTransaction.getAccountId()).isEqualTo(accountId);
+        assertThat(tokenAssociateTransaction.getTokenIds()).isEqualTo(tokenIds);
```

Comment:
C++ SDK tests against size of tokenIds [here](https://github.com/hashgraph/hedera-sdk-cpp/blob/63b7e0300795c2a402308841bfad1fc9ff988f46/sdk/tests/unit/TokenAssociateTransactionUnitTests.cc#L68)

I understand that this option in PR should work as well, but I mean should we make completely matching tests for the sake of consistency between SDK codebases or no?

Reply:
I agree, we should make completely matching tests where it's possible for the `C++` and the `Java` SDKs.
There should be added an assert to check the size of `tokenIds`.